framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -3317,6 +3320,43 @@ private AnnotationMirror getDeclAnnotation(
         return results;
     }
 
+    /**
+     * Adds into {@code results} the declaration annotations found in all elements of the super
+     * types of {@code typeMirror}.
+     */
+    private void inheritOverriddenDeclAnnosFromTypeDecl(
```

Comment:
I hadn't noticed this behavior earlier, but it does mean that "annotated with CollectionType" isn't quite accurate.